src/main/java/com/google/googleidentity/oauth2/token/InMemoryOAuth2TokenService.java

```diff
@@ -0,0 +1,306 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.token;
+
+import com.google.common.collect.ImmutableList;
+import com.google.googleidentity.oauth2.request.OAuth2Request;
+import com.google.inject.Singleton;
+
+import javax.crypto.KeyGenerator;
+import java.security.InvalidParameterException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.UUID;
+import java.util.Optional;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Default in memory token service.
+ * An implementation of {@link OAuth2TokenService}.
+ * The encryption of clientId and username is only needed in this InMemory Design.
+ */
+@Singleton
+public class InMemoryOAuth2TokenService implements OAuth2TokenService {
```

Comment:
Make sure all the methods part of the OAuth2TokenService interface are annotated `@Override`